### ISSUE TITLE
AttributeError: 'Index' object has no attribute 'iterblobs'

### ISSUE THREAD

**Issue by peter17** [arthaud/git-dumper#27]
Hi,

Tanks for your tool, it works very well, in most cases. However, I had the following error:

```
$ gitdumper.py https://(REDACTED)/.git/ repository
[-] Testing https://(REDACTED)/.git/HEAD [200]
[-] Testing https://(REDACTED)/.git/ [403]
[-] Fetching common files
[-] Fetching https://(REDACTED)/.git/COMMIT_EDITMSG [404]
[-] https://(REDACTED)/.git/COMMIT_EDITMSG responded with status code 404
[-] Fetching https://(REDACTED)/.git/hooks/applypatch-msg.sample [200]
[-] Fetching https://(REDACTED)/.git/description [200]
[-] Fetching https://(REDACTED)/.git/hooks/pre-applypatch.sample [200]
[-] Fetching https://(REDACTED)/.git/hooks/post-receive.sample [404]
[-] https://(REDACTED)/.git/hooks/post-receive.sample responded with status code 404
[-] Fetching https://(REDACTED)/.git/hooks/post-commit.sample [404]
[-] https://(REDACTED)/.git/hooks/post-commit.sample responded with status code 404
[-] Fetching https://(REDACTED)/.git/hooks/post-update.sample [200]
[-] Fetching https://(REDACTED)/.git/hooks/pre-commit.sample [200]
[-] Fetching https://(REDACTED)/.gitignore [404]
[-] https://(REDACTED)/.gitignore responded with status code 404
[-] Fetching https://(REDACTED)/.git/hooks/commit-msg.sample [200]
[-] Fetching https://(REDACTED)/.git/hooks/pre-receive.sample [404]
[-] https://(REDACTED)/.git/hooks/pre-receive.sample responded with status code 404
[-] Fetching https://(REDACTED)/.git/hooks/prepare-commit-msg.sample [200]
[-] Fetching https://(REDACTED)/.git/hooks/pre-rebase.sample [200]
[-] Fetching https://(REDACTED)/.git/objects/info/packs [404]
[-] https://(REDACTED)/.git/objects/info/packs responded with status code 404
[-] Fetching https://(REDACTED)/.git/info/exclude [200]
[-] Fetching https://(REDACTED)/.git/hooks/pre-push.sample [200]
[-] Fetching https://(REDACTED)/.git/hooks/update.sample [200]
[-] Fetching https://(REDACTED)/.git/index [200]
[-] Finding refs/
[-] Fetching https://(REDACTED)/.git/info/refs [404]
[-] https://(REDACTED)/.git/info/refs responded with status code 404
[-] Fetching https://(REDACTED)/.git/config [200]
[-] Fetching https://(REDACTED)/.git/ORIG_HEAD [200]
[-] Fetching https://(REDACTED)/.git/logs/refs/stash [404]
[-] https://(REDACTED)/.git/logs/refs/stash responded with status code 404
[-] Fetching https://(REDACTED)/.git/logs/refs/remotes/origin/master [404]
[-] https://(REDACTED)/.git/logs/refs/remotes/origin/master responded with status code 404
[-] Fetching https://(REDACTED)/.git/HEAD [200]
[-] Fetching https://(REDACTED)/.git/FETCH_HEAD [200]
[-] Fetching https://(REDACTED)/.git/packed-refs [200]
[-] Fetching https://(REDACTED)/.git/logs/refs/remotes/origin/HEAD [200]
[-] Fetching https://(REDACTED)/.git/logs/refs/heads/master [404]
[-] https://(REDACTED)/.git/logs/refs/heads/master responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/heads/master [404]
[-] https://(REDACTED)/.git/refs/heads/master responded with status code 404
[-] Fetching https://(REDACTED)/.git/logs/HEAD [200]
[-] Fetching https://(REDACTED)/.git/refs/remotes/origin/HEAD [200]
[-] Fetching https://(REDACTED)/.git/refs/remotes/origin/master [404]
[-] https://(REDACTED)/.git/refs/remotes/origin/master responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/wip/wtree/refs/heads/master [404]
[-] https://(REDACTED)/.git/refs/wip/wtree/refs/heads/master responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/stash [404]
[-] https://(REDACTED)/.git/refs/stash responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/wip/index/refs/heads/master [404]
[-] https://(REDACTED)/.git/refs/wip/index/refs/heads/master responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/heads/develop [200]
[-] Fetching https://(REDACTED)/.git/refs/remotes/origin/develop [404]
[-] https://(REDACTED)/.git/refs/remotes/origin/develop responded with status code 404
[-] Fetching https://(REDACTED)/.git/logs/refs/heads/develop [200]
[-] Fetching https://(REDACTED)/.git/logs/refs/remotes/origin/develop [404]
[-] https://(REDACTED)/.git/logs/refs/remotes/origin/develop responded with status code 404
[-] Fetching https://(REDACTED)/.git/logs/refs/tags/1.0.0 [404]
[-] https://(REDACTED)/.git/logs/refs/tags/1.0.0 responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/tags/1.0.2 [404]
[-] https://(REDACTED)/.git/refs/tags/1.0.2 responded with status code 404
[-] Fetching https://(REDACTED)/.git/logs/refs/tags/1.0.1 [404]
[-] https://(REDACTED)/.git/logs/refs/tags/1.0.1 responded with status code 404
[-] Fetching https://(REDACTED)/.git/logs/refs/tags/1.0.2 [404]
[-] https://(REDACTED)/.git/logs/refs/tags/1.0.2 responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/tags/1.1.0 [404]
[-] https://(REDACTED)/.git/refs/tags/1.1.0 responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/tags/1.0.0 [404]
[-] https://(REDACTED)/.git/refs/tags/1.0.0 responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/tags/1.0.1 [404]
[-] https://(REDACTED)/.git/refs/tags/1.0.1 responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/tags/1.1.1 [404]
[-] https://(REDACTED)/.git/refs/tags/1.1.1 responded with status code 404
[-] Fetching https://(REDACTED)/.git/logs/refs/tags/1.1.0 [404]
[-] https://(REDACTED)/.git/logs/refs/tags/1.1.0 responded with status code 404
[-] Fetching https://(REDACTED)/.git/logs/refs/tags/1.1.1 [404]
[-] https://(REDACTED)/.git/logs/refs/tags/1.1.1 responded with status code 404
[-] Fetching https://(REDACTED)/.git/refs/tags/1.1.2 [404]
[-] Fetching https://(REDACTED)/.git/logs/refs/tags/1.1.2 [404]
[-] https://(REDACTED)/.git/refs/tags/1.1.2 responded with status code 404
[-] https://(REDACTED)/.git/logs/refs/tags/1.1.2 responded with status code 404
[-] Finding packs
[-] Finding objects
Traceback (most recent call last):
  File "/home/peter/bin/gitdumper.py", line 724, in <module>
    main()
  File "/home/peter/bin/gitdumper.py", line 712, in main
    fetch_git(
  File "/home/peter/bin/gitdumper.py", line 571, in fetch_git
    for entry in index.iterblobs():
AttributeError: 'Index' object has no attribute 'iterblobs'
```

Can you please have a look? Thanks a lot!

**Comment by peter17** [comment]
NB: after some investigation, I found that it works if i install dulwich == 0.20.20. I previously had 0.20.30

I have not tried with other versions. You may want to update either your code to work with newer versions of dulwich or `requirements.txt` to lock it to a specific version.

**Comment by bubzzzzzzzz** [comment]
Hi, I am still unable to get this working please can someone help.

┌──(kali㉿kali)-[~]
└─$ git-dumper [http://10.129.96.151/.git](https://urldefense.com/v3/__http://10.129.96.151/.git__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkZILiupM$) ./eplison
[-] Testing [http://10.129.96.151/.git/HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkc927Oro$) [200]
[-] Testing [http://10.129.96.151/.git/](https://urldefense.com/v3/__http://10.129.96.151/.git/__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkRLljIoE$) [403]
[-] Fetching common files
[-] Fetching [http://10.129.96.151/.gitignore](https://urldefense.com/v3/__http://10.129.96.151/.gitignore__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkYDRJngo$) [404]
[-] [http://10.129.96.151/.gitignore](https://urldefense.com/v3/__http://10.129.96.151/.gitignore__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkYDRJngo$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/hooks/commit-msg.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/commit-msg.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkTsRXMi1$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/applypatch-msg.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/applypatch-msg.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkXDxn1r7$) [200]
[-] Fetching [http://10.129.96.151/.git/COMMIT_EDITMSG](https://urldefense.com/v3/__http://10.129.96.151/.git/COMMIT_EDITMSG__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkfP7P2em$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/post-commit.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/post-commit.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkVF3cNDg$) [404]
[-] [http://10.129.96.151/.git/hooks/post-commit.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/post-commit.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkVF3cNDg$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/hooks/pre-rebase.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/pre-rebase.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkav3APC2$) [200]
[-] Fetching [http://10.129.96.151/.git/description](https://urldefense.com/v3/__http://10.129.96.151/.git/description__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkYrwnrNb$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/pre-applypatch.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/pre-applypatch.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkVWe1KQD$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/pre-push.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/pre-push.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkeDlkr8I$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/post-receive.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/post-receive.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkZIJiUhY$) [404]
[-] [http://10.129.96.151/.git/hooks/post-receive.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/post-receive.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkZIJiUhY$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/hooks/update.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/update.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkdZksk06$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/post-update.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/post-update.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkUbIieZe$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/pre-commit.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/pre-commit.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkVYrvlgR$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/pre-receive.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/pre-receive.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkRqQ-rhj$) [200]
[-] Fetching [http://10.129.96.151/.git/objects/info/packs](https://urldefense.com/v3/__http://10.129.96.151/.git/objects/info/packs__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkfwuWMCg$) [404]
[-] [http://10.129.96.151/.git/objects/info/packs](https://urldefense.com/v3/__http://10.129.96.151/.git/objects/info/packs__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkfwuWMCg$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/index](https://urldefense.com/v3/__http://10.129.96.151/.git/index__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkcjWeeVK$) [200]
[-] Fetching [http://10.129.96.151/.git/info/exclude](https://urldefense.com/v3/__http://10.129.96.151/.git/info/exclude__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkR50CRvM$) [200]
[-] Fetching [http://10.129.96.151/.git/hooks/prepare-commit-msg.sample](https://urldefense.com/v3/__http://10.129.96.151/.git/hooks/prepare-commit-msg.sample__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkSHiZ2wn$) [200]
[-] Finding refs/
[-] Fetching [http://10.129.96.151/.git/HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkc927Oro$) [200]
[-] Fetching [http://10.129.96.151/.git/config](https://urldefense.com/v3/__http://10.129.96.151/.git/config__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkXnf5bpj$) [200]
[-] Fetching [http://10.129.96.151/.git/FETCH_HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/FETCH_HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkRgn_Bme$) [404]
[-] [http://10.129.96.151/.git/FETCH_HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/FETCH_HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkRgn_Bme$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/logs/refs/stash](https://urldefense.com/v3/__http://10.129.96.151/.git/logs/refs/stash__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkc2BOYm4$) [404]
[-] [http://10.129.96.151/.git/logs/refs/stash](https://urldefense.com/v3/__http://10.129.96.151/.git/logs/refs/stash__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkc2BOYm4$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/info/refs](https://urldefense.com/v3/__http://10.129.96.151/.git/info/refs__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkU-PPVvG$) [404]
[-] [http://10.129.96.151/.git/info/refs](https://urldefense.com/v3/__http://10.129.96.151/.git/info/refs__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkU-PPVvG$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/logs/refs/heads/master](https://urldefense.com/v3/__http://10.129.96.151/.git/logs/refs/heads/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkbhpQ8HZ$) [200]
[-] Fetching [http://10.129.96.151/.git/refs/heads/master](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/heads/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkd37LlN_$) [200]
[-] Fetching [http://10.129.96.151/.git/ORIG_HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/ORIG_HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkSDCnr9v$) [200]
[-] Fetching [http://10.129.96.151/.git/logs/refs/remotes/origin/HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/logs/refs/remotes/origin/HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkSUxVM1u$) [404]
[-] [http://10.129.96.151/.git/logs/refs/remotes/origin/HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/logs/refs/remotes/origin/HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkSUxVM1u$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/logs/HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/logs/HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkTYxCLc2$) [200]
[-] Fetching [http://10.129.96.151/.git/refs/remotes/origin/HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/remotes/origin/HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkfVuMxxu$) [404]
[-] Fetching [http://10.129.96.151/.git/refs/remotes/origin/master](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/remotes/origin/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkRz6qKsc$) [404]
[-] [http://10.129.96.151/.git/refs/remotes/origin/master](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/remotes/origin/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkRz6qKsc$) responded with status code 404
[-] [http://10.129.96.151/.git/refs/remotes/origin/HEAD](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/remotes/origin/HEAD__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkfVuMxxu$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/logs/refs/remotes/origin/master](https://urldefense.com/v3/__http://10.129.96.151/.git/logs/refs/remotes/origin/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkfigtnuX$) [404]
[-] [http://10.129.96.151/.git/logs/refs/remotes/origin/master](https://urldefense.com/v3/__http://10.129.96.151/.git/logs/refs/remotes/origin/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkfigtnuX$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/refs/stash](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/stash__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkbYH2Iuq$) [404]
[-] [http://10.129.96.151/.git/refs/stash](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/stash__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkbYH2Iuq$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/packed-refs](https://urldefense.com/v3/__http://10.129.96.151/.git/packed-refs__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkSufLqKK$) [404]
[-] [http://10.129.96.151/.git/packed-refs](https://urldefense.com/v3/__http://10.129.96.151/.git/packed-refs__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkSufLqKK$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/refs/wip/wtree/refs/heads/master](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/wip/wtree/refs/heads/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkWYG8YPS$) [404]
[-] [http://10.129.96.151/.git/refs/wip/wtree/refs/heads/master](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/wip/wtree/refs/heads/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkWYG8YPS$) responded with status code 404
[-] Fetching [http://10.129.96.151/.git/refs/wip/index/refs/heads/master](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/wip/index/refs/heads/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkbv37bZb$) [404]
[-] [http://10.129.96.151/.git/refs/wip/index/refs/heads/master](https://urldefense.com/v3/__http://10.129.96.151/.git/refs/wip/index/refs/heads/master__;!!NVedmbVXLvs!R6gpD-VeCxcA4_sIQk7fju4qKJEo5UJWRVVUURC9Xmw0BkkAXjOtROyUkbv37bZb$) responded with status code 404
[-] Finding packs
[-] Finding objects
**Traceback (most recent call last):
  File "/usr/local/bin/git-dumper", line 8, in <module>
    sys.exit(main())
  File "/home/kali/.local/lib/python3.9/site-packages/git_dumper.py", line 724, in main
    fetch_git(
  File "/home/kali/.local/lib/python3.9/site-packages/git_dumper.py", line 583, in fetch_git
    for entry in index.iterblobs():
AttributeError: 'Index' object has no attribute 'iterblobs'**

[git_dumper.txt](https://github.com/arthaud/git-dumper/files/8131372/git_dumper.txt)

**Comment by bubzzzzzzzz** [comment]
Hi, after speaking to Peter this is now working, related to dulwich version.

Fix which Peter provided: 
pip3 install dulwich==0.20.20
sudo pip3 install dulwich==0.20.20

Thanks

**Comment by arthaud** [comment]
Should be fixed with 0e68785456ff9d8f07e42feb34cc33d832879692

**Comment by OxNinja** [comment]
Hello, sorry to reopen this, yes #28 fixed the problem :pray:, but the `pip` package does not include this version of `dulwich` :slightly_smiling_face: 
If installed with `pip install git-dumper`, one should also `pip install dulwich==0.20.20`

**Comment by arthaud** [comment]
This should be fixed in the latest package, 1.0.6.
